File: 29.EXAM PREPARATION_I/Skeletons/JavaScript Skeleton/models/Film.js
const mongoose = require('mongoose');

let filmSchema = mongoose.Schema({
    //TODO: Implement me ...
    /*
    The Film entity holds 5 properties:
•	id – technology-dependent identifier (ObjectID for JavaScript, int for all other technologies)
•	name – non-empty text
•	genre – non-empty text
•	director – non-empty text
•	year – non-null integer


    * title: {type: 'string', required: 'true'},
    comments: {type: 'string', required: 'true'},
     */
    name: {type: 'string', required: 'true'},
    genre: {type: 'string', required: 'true'},
    director: {type: 'string', required: 'true'},
    year: {type: 'number', required: 'true'}

});

let Film = mongoose.model('Film', filmSchema);

module.exports = Film;

File: 29.EXAM PREPARATION_I/Skeletons/C# Skeleton/IMDB/Models/Film.cs
using System.ComponentModel.DataAnnotations;
using System.Web.Mvc;

namespace IMDB.Models
{
    public class Film
    {
        /* The Film entity holds 5 properties:
            •	id – technology-dependent identifier (ObjectID for JavaScript, int for all other technologies)
            •	name – non-empty text
            •	genre – non-empty text
            •	director – non-empty text
            •	year – non-null integer 

         */

        [Key]
        public int Id { get; set; }

        [Required] //not null
        [MinLength(1)]  //non-empty string
        public string Name { get; set; }

        [Required]
        [MinLength(1)]
        public string Genre { get; set; }

        [Required]
        [MinLength(1)]
        public string Director { get; set; }

        [Range(1000, 3000)]
        public int Year { get; set; }
    }
}